specs/ContextMenuRequested.md

```diff
@@ -0,0 +1,477 @@
+                }
+                cm.IsOpen = true;
+            }
+        }, null);
+    };
+```
+# Remarks
+
+# API Notes
+
+# API Details
+ ```cpp
+    interface ICoreWebView2_4;
+    interface ICoreWebView2Environment;
+    interface ICoreWebView2ContextMenuItem;
+    interface ICoreWebView2ContextMenuItemCollection;
+    interface ICoreWebView2ContextMenuRequestedEventArgs;
+    interface ICoreWebView2ContextMenuRequestedEventHandler;
+
+    /// Defines the Context Menu Items
+    [v1_enum]
+    typedef enum COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR {
+        /// Custome type, will include Items brought in by browser extensions
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR_CUSTOM,
+
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR_MISSPELLED,
+
+        /// No Command
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR_NONE,
+
+        /// Page Commands
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR_BACK,
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR_FORWARD,
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR_RELOAD,
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR_STOP,
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR_NEW_WINDOW,
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR_PRINT,
+
+        /// Clipboard Commands
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR_CUT,
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR_COPY,
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR_PASTE,
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR_PASTE_AS_PLAIN_TEXT,
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR_SELECT_ALL,
+
+        /// Saving
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR_SAVE_LINK_AS,
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR_SAVE_IMAGE_AS,
+
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR_INSPECT, 
+        
+    } COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR;
+    
+    /// Indicates the context selected (where the user right clicked)
+    [v1_enum]
+    typedef enum COREWEBVIEW2_SELECTION_CONTEXT {
+
+        /// Indicates that a user right clicked on the page.
+        COREWEBVIEW2_SELECTION_CONTEXT_DEFAULT,
+
+        /// Indicates that a user right clicked on audio.
+        COREWEBVIEW2_SELECTION_CONTEXT_AUDIO,
+
+        /// Indicates that a user right clicked on an image.
+        COREWEBVIEW2_SELECTION_CONTEXT_IMAGE,
+        
+        /// Indicates that a user right clicked on a video
+        COREWEBVIEW2_SELECTION_CONTEXT_VIDEO,
+
+    } COREWEBVIEW2_SELECTION_CONTEXT;
+
+    /// Indicates the menu item type
+    [v1_enum]
+    typedef enum COREWEBVIEW2_CONTEXT_MENU_ITEM_KIND {
+        /// A normal menu item kind, can be enabled or not.
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_KIND_NORMAL,
+
+        /// A checkbox. Can be checked or enabled
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_KIND_CHECKBOX,
+
+        /// A radio button type, can be checked or enabled
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_KIND_RADIO,
+
+        /// A Separator type, has no underlying functionality
```

Comment:
What does 'no underlying functionality mean'? Do you mean it cannot be disabled and cannot be checked?
```suggestion
        /// A separator type, has no underlying functionality
```

Reply:
I will change the wording to be more clear, but I meant it cannot be invoked or checked. Disabling its visibility would still be possible for the end dev in case they don't want a separator in the UI.